core/modules/mysql/MySqlConfig.cc

```diff
@@ -28,118 +28,75 @@
+                         std::string const& socket, std::string const& dbName)
+    : username(username), password(password), port(0), socket(socket), dbName(dbName) {
+    if (socket.empty()) {
+        LOGS(getLogger(), LOG_LVL_FATAL, "MySQL socket undefined");
+        throw std::invalid_argument("MySQL socket undefined");
     }
 }
 
-/// Initializes self from a file. File format: <key>:<value>
-/// To ignore given token, pass "".
-/// To ignore unrecognized tokens, set the flag to false.
-/// This is handy for reading a subset of a file.
-void
-MySqlConfig::initFromFile(std::string const& fName,
-                          std::string const& hostToken,
-                          std::string const& portToken,
-                          std::string const& userToken,
-                          std::string const& passToken,
-                          std::string const& dbNmToken,
-                          std::string const& sockToken,
-                          bool ignoreUnrecognizedTokens) {
-    std::ifstream f;
-    f.open(fName.c_str());
-    if (!f) {
-        std::stringstream s;
-        s << "Failed to open '" << fName << "'";
-        throw s.str();
-    }
-    std::string line;
-    f >> line;
-    while ( !f.eof() ) {
-        int pos = line.find_first_of(':');
-        if ( pos == -1 ) {
-            std::stringstream s;
-            s << "Invalid format, expecting <token>:<value>. "
-              << "File '" << fName << "', line: '" << line << "'";
-            throw s.str();
-        }
-        std::string token = line.substr(0,pos);
-        std::string value = line.substr(pos+1, line.size());
-        if (hostToken != "" and token == hostToken) {
-            this->hostname = value;
-        } else if (portToken != "" and token == portToken) {
-            this->port = atoi(value.c_str());
-            if ( this->port <= 0 ) {
-                std::stringstream s;
-                s << "Invalid port number " << this->port << ". "
-                  << "File '" << fName << "', line: '" << line << "'";
-                throw s.str();
-            }
-        } else if (userToken != "" and token == userToken) {
-            this->username = value;
-        } else if (passToken != "" and token == passToken) {
-            this->password = value;
-        } else if (dbNmToken != "" and token == dbNmToken) {
-            this->dbName = value;
-        } else if (sockToken != "" and token == sockToken) {
-            this->socket = value;
-        } else if (!ignoreUnrecognizedTokens) {
-            std::stringstream s;
-            s << "Unexpected token: '" << token << "' (supported tokens "
-              << "are: " << hostToken << ", " << portToken << ", "
-              << userToken << ", " << passToken << ", " << dbNmToken << ", "
-              << sockToken << ").";
-            throw(s.str());
-        }
-        f >> line;
+bool MySqlConfig::checkConnection() const {
+    lsst::qserv::sql::SqlConnection scn(*this);
```

Comment:
This is very problematic - I believe that our `sql` module depends on `mysql`, here you introduce reverse dependency which makes circular dependency between these two modules. Besides I think `MySqlConfig` is supposed to know anything about making connections, it's a simple class which is a collection of parameters, nothing else, and it's not supposed to make connections. This whole method needs to be moved someplace else (and I'm not sure it's needed at all).